### PR TITLE
Surface the real git stderr instead of a generic error

### PR DESCRIPTION
**Pull request by mazen-salah**
When git wrote to stderr, the scan failed with `errors.New("stderr is not empty")`, so callers and the scan summary lost the actual cause — e.g. a submodule scan reported only `stderr is not empty` instead of the real `fatal: not a git repository: ...`.

Collect the offending stderr lines and include them in the returned error; benign warnings (rename-limit, auto-pack, gc) are still ignored. Adds tests.

### CODE REVIEW

File: sources/stderr_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package sources
```
Comment (zricethezav) [inline sources/stderr_test.go]: @mazen-salah remove test. Stop submitting PRs w/ extraneous test files

Comment (mazen-salah) [reply sources/stderr_test.go]: @zricethezav sorry about that, removed the test. I will stop adding standalone test files going forward. Thanks!

**Comment by mazen-salah** [comment]
@zricethezav just to be clear, these aren't auto-generated spam. I'm using Claude Code to help me find edge cases and real bugs in the codebase, and I read, test, and review every change myself before opening a PR. Happy to slow down or adjust the format to whatever is least noisy for you.
